libs/node-client/src/aggregate-instance.ts

```diff
@@ -61,19 +114,74 @@ export class AggregateInstance<
         },
       });
 
-      for (const event of events) {
-        const eventHandler = this.eventHandlers.get(event.type);
+      await this.digest(events as TEvent[]);
+
+      const commandHandler = this.commandHandlers.get(command.type);
 
-        assert(eventHandler, `event handler for ${event.type} not found`);
+      assert(commandHandler, `command handler for ${command.type} not found`);
 
-        const state = await eventHandler.handle({ state: this._state} as never, event);
+      const generatedEvent = await commandHandler.handle(
+        {
+          state: this._state,
+        } as TContext & { state: TState },
+        command
+      );
 
-        this._state = state;
-        this._version = event.aggregate.version;
+      let lastEvent = null;
+
+      if (Array.isArray(generatedEvent)) {
+        for await (const event of generatedEvent) {
```

Comment:
if the command handler generates multiple events, the version should increment for every event

Reply:
fixed